lightly/utils/scheduler.py

```diff
@@ -28,7 +36,7 @@ def cosine_schedule(
         raise ValueError("Current step number can't be negative")
     if max_steps < 1:
         raise ValueError("Total step number must be >= 1")
-    if step > max_steps:
+    if not period and step > max_steps:
```

Comment:
Could you add a check below that raises a `ValueError` if `period is not None and period <= 0`. We should catch this to avoid division by zero errors and unexpected behavior.